llarp/util/metrics_types.hpp

```diff
@@ -0,0 +1,532 @@
+#ifndef LLARP_METRICS_TYPES_HPP
+#define LLARP_METRICS_TYPES_HPP
+
+#include <util/string_view.hpp>
+#include <util/threading.hpp>
+#include <util/types.hpp>
+
+#include <absl/types/span.h>
+#include <iosfwd>
+
```

Comment:
include what you use:
`absl::optional` (falls through to `std::optional` in modern c++)
`cstring`
`set`
`memory`
to fix non-macOS

Reply:
yeah, just fixing up debian. looks like some of that is running into a fun gcc constexpr bug though.

include what you use is a good move, I'm bad at remembering.

Reply:
Fixed.